### PR TITLE
Adding a simple Multi-Node Pingpong PyTorch Workload

### PR DESCRIPTION
**Pull request by parambole**
- Multi-Node Pingpong PyTorch program
- DockerFile
- SLURM deployment scripts
- GKE deployment scripts

### CODE REVIEW

File: sample_workloads/pingpong/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+## Pingpong
+
+### Building the Docker Container
+
+To build the Docker container 
+
+Run the following command
+
+```
+sudo docker build . -t "us-central1-docker.pkg.dev/<YOUR PROJECT ID>/<ARTIFACT REGISTRY NAME>/litgpt-full:<ADD TAG HERE>"
```
Comment (gkroiz) [inline sample_workloads/pingpong/README.md:24]: should we be using `litgpt-full` tag?

**Review by gkroiz — COMMENTED** [review]
@parambole, please take a look at my comments when you can. Relative to the litgpt workload, do we want users to run this beforehand since it is a simpler workload? How does this workload fit into the user onboarding journey?

File: sample_workloads/pingpong/pingpong.py
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/pingpong.py:1]: @gkroiz No, not sure how `git add` missed it. I have updated it.

File: sample_workloads/pingpong/README.md (hunk 1 above)
Comment (parambole) [reply sample_workloads/pingpong/README.md]: Fixed.

File: sample_workloads/pingpong/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+gcloud config set compute/zone $ZONE
+
+gcloud compute ssh <<Cluster Node Id >> --zone=$ZONE --project=$PROJECT_ID
+```
+
+2. Queue the Workload Job
+
+**Note:** 
+* mv `pingpong_container.sh.example` to `pingpong_container.sh.example`
+* Update the image information `    us-central1-docker.pkg.dev/<YOUR PROJECT ID>/<ARTIFACT REGISTRY NAME>/litgpt-full:<ADD TAG HERE>`
```
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md]: Fixed.

File: sample_workloads/pingpong/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Run the following command
+
+```
+sudo docker build . -t "us-central1-docker.pkg.dev/<YOUR PROJECT ID>/<ARTIFACT REGISTRY NAME>/litgpt-full:<ADD TAG HERE>"
+```
+
+Upload the Artifact to a Registry 
+```
+sudo docker push us-central1-docker.pkg.dev/<YOUR PROJECT ID>/<ARTIFACT REGISTRY NAME>/litgpt-full:<ADD TAG HERE>
```
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md]: Fixed.

File: sample_workloads/pingpong/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
+## Overview
+
+This document provides instructions on 
+* Building a simple Pingpong PyTorch-based workload on A3 using GPUDirectTCPx
+* Running the workload on
```
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md:5]: @gkroiz Over here I wanted to highlight the fact that the same workload can be deployed on both GKE and SLURM - This is to be treated as a reference for anyone looking to transition from SLURM->GKE or vice-versa. 

LMK if that point is not coming across

File: sample_workloads/pingpong/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+Upload the Artifact to a Registry 
+```
+sudo docker push us-central1-docker.pkg.dev/<YOUR PROJECT ID>/<ARTIFACT REGISTRY NAME>/litgpt-full:<ADD TAG HERE>
+```
+
+### Running the workload on SLURM
+
+1. Log into the SLURM node
```
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md:34]: @gkroiz once the SLURM cluster is up and running no additional configuration is required. The scripts to run the job should take of everything. 

PLMK if I am missing something in particular.

**Comment by parambole** [comment]
> @parambole, please take a look at my comments when you can. Relative to the litgpt workload, do we want users to run this beforehand since it is a simpler workload? How does this workload fit into the user onboarding journey?

@gkroiz I have responded to your comments PTAL. 

```
Relative to the litgpt workload, do we want users to run this beforehand since it is a simpler workload
```

Yes, the goal of this workload is for users to get familiar with 
1. Multinode pytorch workload 
2. Containerizing a workload 
3. Deploying it on a SLURM cluster
4. Deploying it on a GKE cluster
5. Also act as a reference for SLURM->GKE and vice versa. 

It can be treated as a warm up example which is easy to understand and debug

**Review by Chris113113 — APPROVED** [review]
Approving w/ nit comments

File: sample_workloads/pingpong/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gkroiz) [reply sample_workloads/pingpong/README.md:5]: Ah, I think my comment was moreso we can just keep "slurm" and "gke" on one line but either way works.

File: sample_workloads/pingpong/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (gkroiz) [reply sample_workloads/pingpong/README.md:34]: +1

**Review by gkroiz — APPROVED** [review]
Overall LGTM, please resolve remaining comments that make sense to adjust in this PR before merging.

File: sample_workloads/pingpong/README.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
+## Overview
+
+This document provides instructions on 
+* Building a simple Pingpong PyTorch-based workload on A3 using GPUDirectTCPx
+* Running the workload on
+    - SLURM
+    - GKE
+
+
+## Pre-Requisites
```
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md:7]: SG.

File: sample_workloads/pingpong/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md]: Done - I have put them in one line.

File: sample_workloads/pingpong/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (parambole) [reply sample_workloads/pingpong/README.md:33]: I have added links to GKE and Deploy an HPC cluster with Slurm in the prerequisite section.

**Comment by parambole** [comment]
> Approving w/ nit comments

Thanks for reviewing.

**Comment by parambole** [comment]
> Overall LGTM, please resolve remaining comments that make sense to adjust in this PR before merging.

Thanks for reviewing.

**Pull request merged**